build-a-test-app.md

```diff
@@ -9,6 +9,33 @@ toc: false
 
 ## Python
 
+~~~ python
+# Import the driver.
+import psycopg2
+
+# Assuming a "bank" database already exists, connect to it.
+conn = psycopg2.connect(database='bank', user='root', host='ROACHs-MacBook-Pro.local', port=26257)
+
+# Open a cursor to perform database operations.
+cur = conn.cursor()
+
+# Create an "accounts" table.
+cur.execute("CREATE TABLE accounts (id INT PRIMARY KEY, balance DECIMAL);")
+
+# Insert two rows into the table.
+cur.execute("INSERT INTO accounts (id, balance) VALUES (1, DECIMAL '1000'),(2, DECIMAL '230.50');")
+
+# Update one row.
+cur.execute("UPDATE accounts SET balance = balance - DECIMAL '5.50' WHERE id = 1;")
+
+# Make the changes to the database persistent.
+conn.commit()
```

Comment:
We are doing everything (creating table, insertion, update) in a single transaction. This pattern encourages an ineffective use of transactions, which should be as small as possible. It doesn't look like we need multi-statement transactions at all in this example, so we could just use the autocommit option http://initd.org/psycopg/docs/connection.html#connection.set_session

Though to have an example of all the useful usage modes, we should also add a pair of statements that really do need to be in a transaction. Perhaps two updates of two account balances, simulating a transfer, with client-side checking of balance, something along the lines of:

```python
conn.set_session(autocommit=false)
# Start a transaction
cur.execute('SELECT balance FROM accounts WHERE id == 1')
<use cur.fetchone() to read the balance>
if balance > 100 {
   cur.execute("UPDATE ..")  # subtract 100 from account 1 balance
   cur.execute("UPDATE.. ") # add 100 from account 2 balance
}
# Finish the transaction
cur.commit()
```

And then maybe follow up with an example of how the same can be done with just a single more complicated statement, see https://github.com/cockroachdb/examples-go/blob/master/bank/sql_bank/main.go#L53